ERCS/erc-7683.md

```diff
@@ -24,26 +24,47 @@ By implementing a standard, cross-chain intents systems can interoperate and sha
 
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
-### CrossChainOrder struct
+### Glossary of Terms
 
-A compliant cross-chain order type MUST be ABI decodable into the `CrossChainOrder` type.
+- **Destination Chain**: the chain where the intent is executed and the user receives funds. Note: intents can involve multiple destination chains.
+- **Filler**: a participant who fulfils a user intent on the destination chain(s) and receives payment as a reward.
+- **Leg**: a portion of the user intent that can be executed independently from others. All legs must be executed for an intent to be considered fulfilled.
+- **Origin chain**: the chain where the user sends funds.
+- **Settlement System**: a system that custodies user deposits, verifies fills, and pays fillers for the purpose of faciliating intents.
+- **Settler**: a contract that implements part of the settlement system on a particular chain.
+- **User**: for the purposes of this document, the user is the end-user who is sending the ERC-7683 order.
+
+### Order structs
+
+A compliant cross-chain order type MUST be ABI decodable into either `GaslessCrossChainOrder` or `OnchainCrossChainOrder` type.
 
 ```solidity
-/// @title CrossChainOrder type
-/// @notice Standard order struct to be signed by swappers, disseminated to fillers, and submitted to settlement contracts
-struct CrossChainOrder {
+/// @title GaslessCrossChainOrder CrossChainOrder type
+/// @notice Standard order struct to be signed by users, disseminated to fillers, and submitted to origin settler contracts
+struct GaslessCrossChainOrder {
 	/// @dev The contract address that the order is meant to be settled by.
 	/// Fillers send this order to this contract address on the origin chain
-	address settlementContract;
+	address originSettler;
 	/// @dev The address of the user who is initiating the swap,
 	/// whose input tokens will be taken and escrowed
-	address swapper;
+	address user;
 	/// @dev Nonce to be used as replay protection for the order
 	uint256 nonce;
 	/// @dev The chainId of the origin chain
-	uint32 originChainId;
-	/// @dev The timestamp by which the order must be initiated
-	uint32 initiateDeadline;
+	uint64 originChainId;
+	/// @dev The timestamp by which the order must be opened
+	uint32 openDeadline;
+	/// @dev The timestamp by which the order must be filled on the destination chain
+	uint32 fillDeadline;
+	/// @dev Arbitrary implementation-specific data
+	/// Can be used to define tokens, amounts, destination chains, fees, settlement parameters,
+	/// or any other order-type specific information
+	bytes orderData;
+}
+
+/// @title OnchainCrossChainOrder CrossChainOrder type
+/// @notice Standard order struct for user-opened orders, where the user is the msg.sender.
+struct OnchainCrossChainOrder {
```

Comment:
Good question. You're right that it isn't much more than just some custom data. It could be bytes, but fillDeadline did seem like a field that's worth being shared. We have gotten feedback that a type field would also be useful here as well. Not super strongly opinionated, but if there are shared fields, it seems useful for the struct to be specified to be consistent with the Gasless one.

I do feel strongly that there should be an onchain open _function_, since the standard should support the (common) use case of the users being the msg.sender.

Reply:
I think the question is how does standardizing this on-chain struct or having a standard open function help us with our goal to make an intent interoperable across different settlement systems? 

Because if the intent is generated onchain, then the user already has to select the contract they want to call, and the orderData required.

Since this onchain struct or function call is not visible to any backend systems- they only see the Open event.
So do solvers/settlement systems care about if this "Open event" was generated through this OnchainOrder struct, or some other custom parameters of a function? 

For example in Across, if the parameters of the deposit function changed, but the Deposit event emitted remained the same, then the whole system would still function correctly right?

Reply:
The idea is that the standard is intended to serve two groups: fillers _and_ order creators. If I create an order and want to put it onchain, I want the function I call to be agnostic to the settlement system so my code stays the same no matter which settlement system I use for each order.